File: laba3.1.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
using namespace std;
double y;
double func1(double x) {
    y = x/3 + 0.67;
    return y;
}

double func2(double x) {
    y = tan(x/2);
    return y;
}

double func3(double x) {
    y = (0.5 * x) - 1.5;
    return y;
}



int main()
{
    cout << "Funktionswert" << endl;
    cout << "X" << "\t" << "Y" << endl;


    double dx = 0.5;
    for (double x = -5; x <= 5; x += dx) {

        if (x >= -5 & x <= -2) {
            cout << fixed << setprecision(2) << x << "  " << func1(x) << endl;
        }
        if (x >= -2 & x <= 2) {
            cout << fixed << setprecision(2) << x << "  " << func2(x) << endl;
        }
        if (x >= 2 & x <= 5) {
            cout << fixed << setprecision(2) << x << "  " << func3(x) << endl;
        }
    }
   return 0;
}

File: laba3.1.go
package main

import (
    "fmt"
    "math"
)

func func1(x float64) float64 {
    y := x/3 + 0.67
    return y
}

func func2(x float64) float64 {
    y := math.Tan(x/2)
    return y
}

func func3(x float64) float64 {
    y := (0.5 * x) - 1.5
    return y
}

func main() {
    fmt.Println("Funktionswert")
    fmt.Println("X\tY")

    dx := 0.5
    for x := -5.0; x <= 5; x += dx {
        if x >= -5 && x <= -2 {
            fmt.Printf("%.2f  %.2f\n", x, func1(x))
        }
        if x >= -2 && x <= 2 {
            fmt.Printf("%.2f  %.2f\n", x, func2(x))
        }
        if x >= 2 && x <= 5 {
            fmt.Printf("%.2f  %.2f\n", x, func3(x))
        }
    }
}

